pkg/agent/manager/cache/cache.go

```diff
@@ -355,11 +360,39 @@ func (c *Cache) UpdateSVIDs(update *UpdateSVIDs) {
 			telemetry.SPIFFEID: record.entry.SpiffeId,
 		})
 		log.Debug("SVID updated")
+
+		// Registration entry is updated, remove it from stale map
+		delete(c.staleEntries, entryID)
 	}
 
 	c.notifyBySelectors(notifySet)
 }
 
+// GetStaleEntries obtains a list of stale entries
+func (c *Cache) GetStaleEntries() map[*common.RegistrationEntry]time.Time {
```

Comment:
How would you feel about returning a slice here instead of a map? I don't anticipate callers wanting to do lookups and it feels more "natural" to me.
```suggestion
type StaleEntry struct {
    Entry *common.RegistrationEntry
    ExpiresAt time.Time
}

func (c *Cache) GetStaleEntries() []StaleEntry {
```

Reply:
yes, that looks like a good idea I was thinking to add an struct so we can use there, I'll refactor it